doc/modules/pipeline.rst

```diff
@@ -124,6 +125,40 @@ i.e. if the last estimator is a classifier, the :class:`Pipeline` can be used
 as a classifier. If the last estimator is a transformer, again, so is the
 pipeline.
 
+.. _cached_pipeline:
+
+CachedPipeline: avoid to repeat computation
+===========================================
+
+.. currentmodule:: sklearn.pipeline
+
+:class:`CachedPipeline` can be used instead of :class:`Pipeline` to avoid to
+compute the fit transformers within a pipeline if the parameters and input data
+are identical. A typical example is the case of a grid search in which the
+transformers can be fitted only once and reuse for each configuration.
+
+Usage
+-----
+
+Similarly to :class:`Pipeline`, the pipeline is built on the same manner. However,
+an extra parameter ``memory`` is needed in order to cache the transformers.
+``memory`` can be either a string containing the directory where to cache the
+transfomers or a ``joblib.Memory`` object::
+
+    >>> from sklearn.decomposition import PCA
+    >>> from sklearn.svm import SVC
+    >>> from sklearn.pipeline import CachedPipeline
+    >>> estimators = [('reduce_dim', PCA()), ('clf', SVC())]
+    >>> pipe = CachedPipeline(estimators)
+    >>> pipe # doctest: +NORMALIZE_WHITESPACE
+    CachedPipeline(memory=Memory(cachedir=None),
+    steps=[('reduce_dim', PCA(copy=True, iterated_power='auto',
```

Comment:
I know that there is the "normalize_whitespace" for the tests, so this will pass, but it would be good to indent this and below for readability (and to give good habits to users)